### PR TITLE
Allows admins to pass parameters to New()

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

**Comment by octareenroon91** [comment]
Can you make a separate verb for this? Currently there are cases where spawn opens no dialogue boxes when called. I'd prefer that behavior preserved

**Comment by kevinz000** [comment]
What he said just do advanced spawn or something.

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -616,14 +616,24 @@
...
+		if(!usr)
+			return
+		
+		arguments = list(usr.loc) + arguments
+
+		var/atom/A = new chosen(argslist(arguments))
 		A.admin_spawned = TRUE
 
-	log_admin("[key_name(usr)] spawned [chosen] at ([usr.x],[usr.y],[usr.z])")
+	log_admin("[key_name(usr)] spawned [chosen] at ([usr.x],[usr.y],[usr.z])[LAZYLEN(arguments) > 1 " with parameters [print_single_line(arguments)]": ""]")
```
Comment (octareenroon91) [inline code/modules/admin/admin.dm]: Travis hates this

Comment (lzimann) [inline code/modules/admin/admin.dm]: Change the coords to use the `COORDS(usr)` define, please. Also forgot the `?` for ternary.

**Comment by octareenroon91** [comment]
````
$ tools/travis/build_byond.sh
DM compiler version 511.1380
loading tgstation.mdme
loading interface/skin.dmf
code/modules/admin/admin.dm:626:error: get_callproc_args: undefined proc
code/modules/admin/admin.dm:633:error: argslist: undefined proc
code/modules/admin/admin.dm:636:error: COORDS: undefined proc
code/modules/admin/verbs/modifyvariables.dm:223:error: argslist: undefined proc
code/modules/admin/verbs/modifyvariables.dm:237:error: argslist: undefined proc
code/modules/admin/verbs/modifyvariables.dm:260:error: argslist: undefined proc
tgstation.mdme.dmb - 6 errors, 0 warnings (4/29/17 10:50 pm)
mv: cannot stat `tgstation.mdme.dmb': No such file or directory
````

now what

~~~~~~
`__DEFINES/misc.dm:#define COORD(A) "([A.x],[A.y],[A.z])"`
`arglist()`

This one is probably a dme sequence thing:
`modules/admin/verbs/debug.dm:/client/proc/get_callproc_args()`

**Comment by octareenroon91** [comment]
Also please make this a distinct verb or add a per-admin pref on it.

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -610,20 +610,41 @@
...
 	set name = "Spawn"
 
+	return spawn_atom_impl(object, FALSE)
+
+/datum/admins/proc/spawn_atom_adv(object as text)
+	set category = "Debug"
+	set desc = "(atom path) Spawn an atom with New() parameters"
+	set name = "Advanced Spawn"
+
+	return spawn_atom_impl(object, TRUE)
```
Comment (octareenroon91) [inline code/modules/admin/admin.dm]: I'd probably have requested input here, but whatever works is fine.

**Review by octareenroon91 — APPROVED** [review]

**Pull request merged**